### ISSUE TITLE
AWS Java SDK is throwing different exceptions than the expected [SES]

### ISSUE THREAD

**Issue by dheerajkulakarni** [localstack/localstack#3466]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
AWS Java SDK is  throwing different exceptions than the expected
version: 1.11.327

**Example 1:**  send a notification with an unverified email address as from address follows

## Expected behavior
exception which is thrown by AWS is like 
com.amazonaws.services.simpleemail.model.MessageRejectedException: Email address is not verified. The following identities failed the check in region US-EAST-1: arn:aws:ses:us-east-1:654856854863:identity/miqdigitl.com (Service: AmazonSimpleEmailService; Status Code: 400; Error Code: MessageRejected; Request ID: 0632aab8-0be1-4eeb-9d0f-8c699330aabf) 

## Actual behavior
 exception thrown when using localstack is like
com.amazonaws.services.simpleemail.model.AmazonSimpleEmailServiceException: null (Service: AmazonSimpleEmailService; Status Code: 400; Error Code: 400 ; Request ID: null)

**Example 2 :** send create template request

## Expected behavior
Template gets created successfully

## Actual behavior
com.amazonaws.services.simpleemail.model.AmazonSimpleEmailServiceException: null (Service: AmazonSimpleEmailService; Status Code: 500; Error Code: 500 ; Request ID: null)

# Steps to reproduce
1. use AWS java SDK in java project
2. make localstack up using docker latest image
3. try to send with unverified email address/ create template connecting to localstack 

## Command used to start LocalStack
docker run --env SERVICES=ses  -p 4566:4566  localstack/localstack

## Client code 

**in kotlin**
val endpoint : AwsClientBuilder.EndpointConfiguration = AwsClientBuilder
      .EndpointConfiguration("http://localhost:4566","us-east-1")
val client = AmazonSimpleEmailServiceClientBuilder.standard()
          .withEndpointConfiguration(endpointConfiguration)
          .build()
client.createTemplate(templateRequest)
         or 
client.sendEmail(sendEmailRequest)

please let me know if I am doing something wrong here as well

**Comment by whummer** [comment]
Thanks for reporting @dheerajkulakarni . This should be fixed in #3491 - can you please pull the `latest` image and give it another try? Please report here if the problem persists. Thanks

**Comment by dheerajkulakarni** [comment]
Hey @whummer, problem which I mentioned in the second scenario about the template creation still persists, first one got solved, can you please verify this.
cc: @pinzon

**Comment by pinzon** [comment]
Hi @dheerajkulakarni.
You're right, I assumed It was the same fix for both issues. 
I'm working on the patch right now.
Thank you.

**Comment by dheerajkulakarni** [comment]
Hey @pinzon, glad you picked this up again, from what I observed it's not only during the creation but there are other issues as well with templates, one thing I am facing right now is, I have created a template using CLI which worked perfectly, and when I use that template to send mail through SDK then again it was giving me 500 error. Not sure when exactly aws throws 500 error, is it something related to unsupported operation error/unable to serve this operation?  if yes then I think we will get this in all operations which include templates right?

**Comment by pinzon** [comment]
@dheerajkulakarni can you give me more details about your case? I just tried to reproduce your issue and for me it seems to work fine. 

This is how I do it:

1.- create file template.json:

```
{
  "Template": {
    "TemplateName": "MyTemplate",
    "SubjectPart": "Greetings, {{name}}!",
    "HtmlPart": "<h1>Hello {{name}},</h1><p>Your favorite animal is {{favoriteanimal}}.</p>",
    "TextPart": "Dear {{name}},\r\nYour favorite animal is {{favoriteanimal}}."
  }
}
```

2.- create template with awscli:
`awslocal ses create-template --cli-input-json file://template.json`

3.- send templated email with java aws-sdk:
```
        AmazonSimpleEmailService client = AmazonSimpleEmailServiceClientBuilder.standard()
                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("http://localhost:4566", "us-east-1"))
                .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials("test", "test")))
                .withClientConfiguration(new ClientConfiguration())
                .build();
        VerifyEmailAddressRequest verifyEmailAddressRequest = new VerifyEmailAddressRequest().withEmailAddress(TO);
        client.verifyEmailAddress(verifyEmailAddressRequest);
        
        verifyEmailAddressRequest.setEmailAddress(FROM);
        client.verifyEmailAddress(verifyEmailAddressRequest);

        // this.testCreateTemplate();

        SendTemplatedEmailRequest request = new SendTemplatedEmailRequest()
                .withSource(FROM).withConfigurationSetName(CONFIGSET)
                .withDestination(new Destination().withToAddresses(TO))
                .withTemplate("MyTemplate")
                .withTemplateData( "{ \"name\":\"Alejandro\", \"favoriteanimal\": \"alligator\" }");
                ;
                
        SendTemplatedEmailResult result = client.sendTemplatedEmail(request);
        Assert.assertNotNull(result);
```

Thank you for your help.

**Comment by dheerajkulakarni** [comment]
Sorry my bad @pinzon , I was actually doing **client.testRenderTemplate(testRenderTemplateRequest)** before doing **client.sendTemplatedEmail(request)** to check if all the required template variables are present, so this was throwing 500 error for me.

// code snippet
val testRenderTemplateRequest = TestRenderTemplateRequest()
testRenderTemplateRequest.templateData = templateVariablesData
testRenderTemplateRequest.templateName = templateId
client.testRenderTemplate(testRenderTemplateRequest)

so basically client.createTemplate() and client.testRenderTemplate() , these are two functions which are throwing 500 errors.

Thanks!

**Comment by dheerajkulakarni** [comment]
Hey @pinzon , Just wanted to know if you are able to reproduce the same behaviour 
cc: @whummer 

Thanks!!

**Comment by pinzon** [comment]
Hi @dheerajkulakarni.
Yes, I was able to reproduce this issue. The crash is caused because the test_render_template function hasn't been implemented in moto.

So we're working on implementing it.

**Comment by dheerajkulakarni** [comment]
Thanks for the update @pinzon !

**Comment by dheerajkulakarni** [comment]
Hey @pinzon, seems like the edit template function is also not implemented.

This is not working through the client and also through CLI 

**through jdk** : client.updateTeamnplate(updateTemplateRequest) , **Result** : No response getting struck in update template call
**through cli** :  aws ses update-template --cli-input-json file://template.json --region=us-east-1 --endpoint-url=http://localhost:4566 , Result: The server encountered an internal error and was unable to complete your request

template.json
{
  "Template": {
    "TemplateName": "test",
    "SubjectPart": "testCount",
    "HtmlPart": "<html><body>{{body}}</body></html>",
    "TextPart": ""
  }
}


Let me know if I can contribute to this in localstack or moto , so that I can try to implement this myself and I can raise a  PR, if yes can you please guide me with some Quickstart. Thanks!

**Comment by pinzon** [comment]
Hi @dheerajkulakarni , Thank you for your interest. 
I just finished adding the function test_render_template to my moto fork. 
I will add at once that function (it shouldn't be as hard as render_template was)
It shouldn't take long.

Ps here is the branch where I'm pushing my progress if you're interested: https://github.com/pinzon/moto/tree/ses-validation-render

**Comment by pinzon** [comment]
Hi @dheerajkulakarni my PR has just been merged. It shouldn't take long to see the changes in Localstack.
I'll close this issue for now. Feel free to comment here or open another issue if there are more problems.
Thanks.

**Comment by dheerajkulakarni** [comment]
Thanks @pinzon for taking your time and solving this !!
Hey @whummer, can you please let me know if this got pushed to local stack latest image so that I can test these changes once in my local.
Thanks!

**Comment by dheerajkulakarni** [comment]
Hey @whummer, Any chance that these changes got pushed to the local stack image?
